diff --git a/dist/chunk-E4IYT7KJ.js b/dist/chunk-E4IYT7KJ.js
index e84eec66521928da901fdad9771f9ec133e584e6..6e7eda5cef77dfdaaa0694f8b24ec78fb9d7fa1b 100644
--- a/dist/chunk-E4IYT7KJ.js
+++ b/dist/chunk-E4IYT7KJ.js
@@ -1,6 +1,6 @@
 // ../generouted/src/core.ts
 var patterns = {
-  route: [/^.*\/src\/pages\/|\.(jsx|tsx|mdx)$/g, ""],
+  route: [/^.*\/src\/routes\/|\.(jsx|tsx|mdx)$/g, ""],
   splat: [/\[\.{3}\w+\]/g, "*"],
   param: [/\[([^\]]+)\]/g, ":$1"],
   slash: [/^index$|\./g, "/"],
diff --git a/dist/index-lazy.js b/dist/index-lazy.js
index 7d85bfab942aab09f31ecc928ed8f4685f3a959b..858ca360beb6366aa4468aa742cb924fca1d680a 100644
--- a/dist/index-lazy.js
+++ b/dist/index-lazy.js
@@ -8,13 +8,13 @@ import {
 import { createComponent as _$createComponent } from "solid-js/web";
 import { createMemo, ErrorBoundary, lazy, Show, Suspense } from "solid-js";
 import { Router, useLocation } from "@solidjs/router";
-var PRESERVED = import.meta.glob("/src/pages/(_app|404).{jsx,tsx}", {
+var PRESERVED = import.meta.glob("/src/routes/(_app|404).{jsx,tsx}", {
   eager: true
 });
-var MODALS = import.meta.glob("/src/pages/**/[+]*.{jsx,tsx}", {
+var MODALS = import.meta.glob("/src/routes/**/[+]*.{jsx,tsx}", {
   eager: true
 });
-var ROUTES = import.meta.glob(["/src/pages/**/[\\w[-]*.{jsx,tsx,mdx}", "!/src/pages/**/(_!(layout)*(/*)?|_app|404)*"]);
+var ROUTES = import.meta.glob(["/src/routes/**/[\\w[-]*.{jsx,tsx,mdx}", "!/src/routes/**/(_!(layout)*(/*)?|_app|404)*"]);
 var preservedRoutes = generatePreservedRoutes(PRESERVED);
 var modalRoutes = generateModalRoutes(MODALS);
 var regularRoutes = generateRegularRoutes(ROUTES, (module) => {
diff --git a/dist/index.js b/dist/index.js
index 957f46636be859cd01259b4e14f98b66c1667f0f..9a2cee5a0f37832eadd869bcf0988e7daecd6a3a 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -8,13 +8,13 @@ import {
 import { createComponent as _$createComponent } from "solid-js/web";
 import { createMemo, ErrorBoundary, Show, Suspense } from "solid-js";
 import { Router, useLocation } from "@solidjs/router";
-var PRESERVED = import.meta.glob("/src/pages/(_app|404).{jsx,tsx}", {
+var PRESERVED = import.meta.glob("/src/routes/(_app|404).{jsx,tsx}", {
   eager: true
 });
-var MODALS = import.meta.glob("/src/pages/**/[+]*.{jsx,tsx}", {
+var MODALS = import.meta.glob("/src/routes/**/[+]*.{jsx,tsx}", {
   eager: true
 });
-var ROUTES = import.meta.glob(["/src/pages/**/[\\w[-]*.{jsx,tsx,mdx}", "!/src/pages/**/(_!(layout)*(/*)?|_app|404)*"], {
+var ROUTES = import.meta.glob(["/src/routes/**/[\\w[-]*.{jsx,tsx,mdx}", "!/src/routes/**/(_!(layout)*(/*)?|_app|404)*"], {
   eager: true
 });
 var preservedRoutes = generatePreservedRoutes(PRESERVED);
diff --git a/dist/plugin/chunk-E4HTHNTM.js b/dist/plugin/chunk-E4HTHNTM.js
index 7fce419e45916cdee15629efed73823bffe2e81a..5103419b309c49c06dc6dc88343004e07c12e389 100644
--- a/dist/plugin/chunk-E4HTHNTM.js
+++ b/dist/plugin/chunk-E4HTHNTM.js
@@ -10,7 +10,7 @@ import { createLogger } from "vite";
 
 // ../../shared/core/src/index.ts
 var patterns = {
-  route: [/^.*\/src\/pages\/|^\/pages\/|\.(jsx|tsx|mdx)$/g, ""],
+  route: [/^.*\/src\/routes\/|^\/routes\/|\.(jsx|tsx|mdx)$/g, ""],
   splat: [/\[\.{3}\w+\]/g, "*"],
   param: [/\[([^\]]+)\]/g, ":$1"],
   slash: [/^index$|\./g, "/"],
@@ -20,8 +20,8 @@ var patterns = {
 // src/plugin/generate.ts
 import fg from "fast-glob";
 var generateRouteTypes = async (options) => {
-  const files = await fg(options.source.routes || "./src/pages/**/[\\w[-]*.{jsx,tsx,mdx}", { onlyFiles: true });
-  const modal = await fg(options.source.modals || "./src/pages/**/[+]*.{jsx,tsx,mdx}", { onlyFiles: true });
+  const files = await fg(options.source.routes || "./src/routes/**/[\\w[-]*.{jsx,tsx,mdx}", { onlyFiles: true });
+  const modal = await fg(options.source.modals || "./src/routes/**/[+]*.{jsx,tsx,mdx}", { onlyFiles: true });
   const filtered = files.filter((key) => !key.includes("/_") && !key.includes("/404"));
   const params = [];
   const paths = filtered.map((key) => {
diff --git a/dist/plugin/chunk-ETY5IWEB.js b/dist/plugin/chunk-ETY5IWEB.js
index fecad09b290d9c9c86347cb1e7a21cdf6cf0b535..5e6c1edc89e845c4511f3b71ca79553c0ec98bc7 100644
--- a/dist/plugin/chunk-ETY5IWEB.js
+++ b/dist/plugin/chunk-ETY5IWEB.js
@@ -1,6 +1,6 @@
 // src/plugin/options.ts
 var defaultOptions = {
-  source: { routes: "./src/pages/**/[\\w[-]*.{jsx,tsx,mdx}", modals: "./src/pages/**/[+]*.{jsx,tsx,mdx}" },
+  source: { routes: "./src/routes/**/[\\w[-]*.{jsx,tsx,mdx}", modals: "./src/routes/**/[+]*.{jsx,tsx,mdx}" },
   output: "./src/router.ts",
   format: true
 };
diff --git a/dist/plugin/generate.cjs b/dist/plugin/generate.cjs
index 9834975e7b69dfbfaec740c51150a62f97b0d55e..45283158aec54079201057aa4990096121452348 100644
--- a/dist/plugin/generate.cjs
+++ b/dist/plugin/generate.cjs
@@ -40,7 +40,7 @@ var import_vite = require("vite");
 
 // ../../shared/core/src/index.ts
 var patterns = {
-  route: [/^.*\/src\/pages\/|^\/pages\/|\.(jsx|tsx|mdx)$/g, ""],
+  route: [/^.*\/src\/routes\/|^\/routes\/|\.(jsx|tsx|mdx)$/g, ""],
   splat: [/\[\.{3}\w+\]/g, "*"],
   param: [/\[([^\]]+)\]/g, ":$1"],
   slash: [/^index$|\./g, "/"],
@@ -64,8 +64,8 @@ export const { useMatch, useModals, useNavigate, useParams } = hooks<Path, Param
 
 // src/plugin/generate.ts
 var generateRouteTypes = async (options) => {
-  const files = await (0, import_fast_glob.default)(options.source.routes || "./src/pages/**/[\\w[-]*.{jsx,tsx,mdx}", { onlyFiles: true });
-  const modal = await (0, import_fast_glob.default)(options.source.modals || "./src/pages/**/[+]*.{jsx,tsx,mdx}", { onlyFiles: true });
+  const files = await (0, import_fast_glob.default)(options.source.routes || "./src/routes/**/[\\w[-]*.{jsx,tsx,mdx}", { onlyFiles: true });
+  const modal = await (0, import_fast_glob.default)(options.source.modals || "./src/routes/**/[+]*.{jsx,tsx,mdx}", { onlyFiles: true });
   const filtered = files.filter((key) => !key.includes("/_") && !key.includes("/404"));
   const params = [];
   const paths = filtered.map((key) => {
diff --git a/dist/plugin/index.cjs b/dist/plugin/index.cjs
index 0463c6f8a32446539eefdd2c586d39f6e19cda3f..27cb224f6a17a2564d2a23a6396acd2bc9a0754d 100644
--- a/dist/plugin/index.cjs
+++ b/dist/plugin/index.cjs
@@ -43,7 +43,7 @@ var import_vite = require("vite");
 
 // ../../shared/core/src/index.ts
 var patterns = {
-  route: [/^.*\/src\/pages\/|^\/pages\/|\.(jsx|tsx|mdx)$/g, ""],
+  route: [/^.*\/src\/routes\/|^\/routes\/|\.(jsx|tsx|mdx)$/g, ""],
   splat: [/\[\.{3}\w+\]/g, "*"],
   param: [/\[([^\]]+)\]/g, ":$1"],
   slash: [/^index$|\./g, "/"],
@@ -67,8 +67,8 @@ export const { useMatch, useModals, useNavigate, useParams } = hooks<Path, Param
 
 // src/plugin/generate.ts
 var generateRouteTypes = async (options) => {
-  const files = await (0, import_fast_glob.default)(options.source.routes || "./src/pages/**/[\\w[-]*.{jsx,tsx,mdx}", { onlyFiles: true });
-  const modal = await (0, import_fast_glob.default)(options.source.modals || "./src/pages/**/[+]*.{jsx,tsx,mdx}", { onlyFiles: true });
+  const files = await (0, import_fast_glob.default)(options.source.routes || "./src/routes/**/[\\w[-]*.{jsx,tsx,mdx}", { onlyFiles: true });
+  const modal = await (0, import_fast_glob.default)(options.source.modals || "./src/routes/**/[+]*.{jsx,tsx,mdx}", { onlyFiles: true });
   const filtered = files.filter((key) => !key.includes("/_") && !key.includes("/404"));
   const params = [];
   const paths = filtered.map((key) => {
@@ -114,7 +114,7 @@ var generate = async (options) => {
 
 // src/plugin/options.ts
 var defaultOptions = {
-  source: { routes: "./src/pages/**/[\\w[-]*.{jsx,tsx,mdx}", modals: "./src/pages/**/[+]*.{jsx,tsx,mdx}" },
+  source: { routes: "./src/routes/**/[\\w[-]*.{jsx,tsx,mdx}", modals: "./src/routes/**/[+]*.{jsx,tsx,mdx}" },
   output: "./src/router.ts",
   format: true
 };
@@ -126,7 +126,7 @@ function Generouted(options) {
     name: "generouted/solid-router",
     enforce: "pre",
     configureServer(server) {
-      const listener = (file = "") => file.includes(import_path2.default.normalize("/src/pages/")) ? generate(resolvedOptions) : null;
+      const listener = (file = "") => file.includes(import_path2.default.normalize("/src/routes/")) ? generate(resolvedOptions) : null;
       server.watcher.on("add", listener);
       server.watcher.on("change", listener);
       server.watcher.on("unlink", listener);
diff --git a/dist/plugin/index.js b/dist/plugin/index.js
index 2f913aa95895fc5a3be2e02f7df9dd8ea8018cec..b624754b26d31c7239bdf7f08fbe342fd365a069 100644
--- a/dist/plugin/index.js
+++ b/dist/plugin/index.js
@@ -14,7 +14,7 @@ function Generouted(options) {
     name: "generouted/solid-router",
     enforce: "pre",
     configureServer(server) {
-      const listener = (file = "") => file.includes(path.normalize("/src/pages/")) ? generate(resolvedOptions) : null;
+      const listener = (file = "") => file.includes(path.normalize("/src/routes/")) ? generate(resolvedOptions) : null;
       server.watcher.on("add", listener);
       server.watcher.on("change", listener);
       server.watcher.on("unlink", listener);
diff --git a/dist/plugin/options.cjs b/dist/plugin/options.cjs
index 15e07247bf398007d84f38f578b1661819907ff7..3d69501addc2f52fb8983a20978d35652fc18a57 100644
--- a/dist/plugin/options.cjs
+++ b/dist/plugin/options.cjs
@@ -24,7 +24,7 @@ __export(options_exports, {
 });
 module.exports = __toCommonJS(options_exports);
 var defaultOptions = {
-  source: { routes: "./src/pages/**/[\\w[-]*.{jsx,tsx,mdx}", modals: "./src/pages/**/[+]*.{jsx,tsx,mdx}" },
+  source: { routes: "./src/routes/**/[\\w[-]*.{jsx,tsx,mdx}", modals: "./src/routes/**/[+]*.{jsx,tsx,mdx}" },
   output: "./src/router.ts",
   format: true
 };
diff --git a/readme.md b/readme.md
index 7e383a9a554b7fb3fe115cea63d784af96239244..9e2922bd84583556b84a8425b3adfb312281d0ba 100644
--- a/readme.md
+++ b/readme.md
@@ -6,7 +6,7 @@ Check out `generouted`'s main [docs](/) for the [features](/#features), [convent
 
 ## How
 
-This integration is based on a Vite plugin to generate routes types for Solid Router with `generouted` conventions. The output is saved by default at `src/router.ts` and gets updated by the add/change/delete at `src/pages/*`.
+This integration is based on a Vite plugin to generate routes types for Solid Router with `generouted` conventions. The output is saved by default at `src/router.ts` and gets updated by the add/change/delete at `src/routes/*`.
 
 ## Getting started
 
@@ -44,10 +44,10 @@ render(Routes, document.getElementById('root')!)
 
 ### Adding pages
 
-Add the home page by creating a new file `src/pages/index.tsx` **→** `/`, then export a default component:
+Add the home page by creating a new file `src/routes/index.tsx` **→** `/`, then export a default component:
 
 ```tsx
-// src/pages/index.tsx
+// src/routes/index.tsx
 
 export default function Home() {
   return <h1>Home</h1>
@@ -57,7 +57,7 @@ export default function Home() {
 ### Optional root layout at `pages/_app.tsx`
 
 ```tsx
-// src/pages/_app.tsx
+// src/routes/_app.tsx
 
 import { ParentProps } from 'solid-js'
 
@@ -79,7 +79,7 @@ export default function App(props: ParentProps) {
 Auto-completion for `A`, `useNavigate`, `useParams` and more exported from `src/router.ts`
 
 ```tsx
-// src/pages/index.tsx
+// src/routes/index.tsx
 import { A, useNavigate, useParams } from '../router'
 
 export default function Home() {
@@ -118,7 +118,7 @@ Although all modals are global, it's nice to co-locate modals with relevant rout
 Create modal routes by prefixing a valid route file name with a plus sign `+`. Why `+`? You can think of it as an extra route, as the modal overlays the current route:
 
 ```tsx
-// src/pages/+login.tsx
+// src/routes/+login.tsx
 
 import { Modal } from '@/ui'
 
@@ -130,7 +130,7 @@ export default function Login() {
 To navigate to a modal use `useModals` hook exported from `src/router.ts`:
 
 ```tsx
-// src/pages/_app.tsx
+// src/routes/_app.tsx
 
 import { ParentProps } from 'solid-js'
 
